aptos-move/move-examples/mint_nft/sources/minting.move

```diff
@@ -0,0 +1,277 @@
+/// This module is an example of how one can create a NFT collection from a resource account
+/// and allow users to mint from the NFT collection.
+/// Check aptos/move-e2e-tests/src/tests /mint.nft.rs for an e2e example.
+///
+/// The basic flow: the module publisher will call `init_module` to initialize the module, and then a nft receiver
+/// can call `mint_nft` to mint a token. If needed, we can update the expiration time of minting using `set_timestamp`
+/// and update if we want to enable token minting using `set_minting_enabled`.
+///
+/// More details:
+/// init_module()
+/// The module publisher will call `init_module` to create a resource account and store the resource account's
+/// address in the publisher's `ModuleData`.
+/// A NFT collection and the first token_data_id to be minted will be created using this resource account.
+/// We then store all the relevant information about this token in `CollectionTokenMinter`.
+///
+/// mint_nft()
+/// `mint_nft` will be called when a user wants to mint a token from this token data id. The user needs to provide
+/// a `mint_proof_signature` (their signature on a MintProofChallenge) to prove that they intend to claim the token.
+/// After function `verify_proof_of_knowledge` verifies that the signature is valid, the signer of the resource account
+/// will transfer the given token_id to the user.
+/// After a token is minted, we call `mutate_token_properties` to update the property version of this token.
+
+module mint_nft::minting {
+    use std::error;
+    use std::signer;
+    use std::string::{Self, String};
+    use std::vector;
+
+    use aptos_framework::account;
+    use aptos_framework::event::{Self, EventHandle};
+    use aptos_framework::timestamp;
+    use aptos_std::ed25519;
+    use aptos_token::token::{Self, TokenDataId};
+
+    // This struct stores the token receiver's address and token_data_id in the event of token minting
+    struct TokenMintingEvent has drop, store {
+        token_receiver_address: address,
+        token_data_id: TokenDataId,
+    }
+
+    // This struct stores an NFT collection's relevant information
+    struct CollectionTokenMinter has key {
+        public_key: ed25519::ValidatedPublicKey,
+        signer_cap: account::SignerCapability,
+        token_data_id: TokenDataId,
+        expiration_timestamp: u64,
+        minting_enabled: bool,
+        token_minting_events: EventHandle<TokenMintingEvent>,
+    }
+
+    // This struct stores the challenge message that proves that the user owns the public key and intends to claims this specific NFT
+    struct MintProofChallenge has drop {
+        sequence_number: u64,
```

Comment:
probably should put the account address of the entity here as well.

Reply:
I am assuming that you meant the account address of the collection token creator? that address is already part of `TokenDataId`